gslib/utils/cloud_api_helper.py

```diff
@@ -67,6 +72,10 @@ def GetDownloadSerializationData(
   if user_project:
     url = AddQueryParamToUrl(url, 'userProject', user_project)
 
+  if six.PY3:
+    if isinstance(url, bytes):
+      url = url.decode('ascii')
```

Comment:
Let's use utf-8 for this, in case the URL has any non-url-encoded utf-8 chars in it.

Reply:
I tried using UTF8 and I started to get failures on tests that are testing utf8 characters. If we change it here, it could require further digging to get it to a useable state again.